vunit/project.py

```diff
@@ -679,7 +680,8 @@ def __init__(self, name, library, file_type):
         self.design_units = []
         self._content_hash = None
         self._compile_options = {}
-
+        content = ostools.read_file(name, encoding=HDL_FILE_ENCODING)
```

Comment:
The old code used file_content_hash (in a way), but I want to avoid this, as for VerilogSourceFile the file_content_hash is modified during parsing. I would prefer something that is not dependent on the parsing so the one can check if the file has to be added without parsing it. Maybe it is a option to cache the file hash additionally to the content hash?

Reply:
With file_content_hash I refer to the function not the content_hash field. The function just returns the hash of the contents of the file and is equivalent with calling hash_string. The only difference is that it does not recompute the hash if the timestamp has not changed.

Reply:
I've tried this. But as I understand it the file_content_hash is dependent on the content_hash field (when pickeled). So when comparing Verilog file_content_hashes they are not equal if they are not parsed, as the _content_hash is modified by the parsing.

Reply:
The content_hash field on the source files are not related to the caching mechanism used by file_content_hash.